File: src/main/java/com/dingdongdeng/coinautotrading/common/type/OrderType.java
package com.dingdongdeng.coinautotrading.common.type;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum OrderType {
    BUY("매수"),
    SELL("매도"),
    CANCEL("주문 취소"),
    ;

    private String desc;
}

File: app-service/src/main/kotlin/com/dingdongdeng/autotrading/infra/common/type/OrderType.kt
package com.dingdongdeng.autotrading.infra.common.type

enum class OrderType(
    override val desc: String,
) : DescriptionType {
    BUY("매수"),
    SELL("매도"),
    CANCEL("주문 취소");

    fun isBuy(): Boolean = this == BUY
    fun isSell(): Boolean = this == SELL
    fun isCancel(): Boolean = this == CANCEL
}
